### PR TITLE
feat: attach .js extension to all relative/absolute file imports/exports

### PR DESCRIPTION
**Pull request by n1ru4l**
Closes https://github.com/dotansimha/graphql-code-generator/issues/8022
Closes https://github.com/dotansimha/graphql-code-generator/pull/8034

### CODE REVIEW

**Comment by gilgardosh** [comment]
@n1ru4l  Is the extension needed here also?
https://github.com/dotansimha/graphql-code-generator/blob/1479233dfa46f576a02b7dff37fd6a52e7aff80a/dev-test/gql-tag-operations-urql/gql/gql.d.ts#L7-L13

File: dev-test/gql-tag-operations/gql/index.ts [hunk 1]
```diff
@@ -1 +1 @@
-export * from './gql';
+export * from './gql.js';
```
Comment (gilgardosh) [inline dev-test/gql-tag-operations/gql/index.ts:1]: I believe it's not needed here

Comment (n1ru4l) [reply dev-test/gql-tag-operations/gql/index.ts:1]: It is needed everywhere actually.

**Comment by n1ru4l** [comment]
> @n1ru4l Is the extension needed here also?
> 
> https://github.com/dotansimha/graphql-code-generator/blob/1479233dfa46f576a02b7dff37fd6a52e7aff80a/dev-test/gql-tag-operations-urql/gql/gql.d.ts#L7-L13

Good catch, fixed it in https://github.com/dotansimha/graphql-code-generator/pull/8034/commits/985d167643e1fba396de1c5fd969a2e03eb54d13

**Comment by gilgardosh** [comment]
this should be rethinked:
https://github.com/dotansimha/graphql-code-generator/blob/d84afec0950b234cd716ea76436e3800a978328c/packages/presets/graphql-modules/src/index.ts#L72

extension are removed, for example
https://github.com/dotansimha/graphql-code-generator/blob/9a5f31cb622f064db58e27962996235d4cf89000/dev-test/codegen.yml#L360
resolves into
https://github.com/dotansimha/graphql-code-generator/blob/faa13973e0022215b57699636a886d70b8b4946e/dev-test/modules/blog/generated.ts#L1

**Review by dotansimha — COMMENTED** [review]
@n1ru4l do we have any concerns about breaking changes? older TS version or anything like that? (in the generated files, i mean)

**Comment by gilgardosh** [comment]
https://github.com/dotansimha/graphql-code-generator/blob/d84afec0950b234cd716ea76436e3800a978328c/packages/plugins/typescript/oclif/src/visitor.ts#L34
default needs extension

**Comment by gilgardosh** [comment]
https://github.com/dotansimha/graphql-code-generator/blob/d84afec0950b234cd716ea76436e3800a978328c/packages/plugins/typescript/resolvers/tests/ts-resolvers.spec.ts#L239

also, in this file, look for assurances of `../entities`

**Comment by n1ru4l** [comment]
> graphql-code-generator/packages/plugins/typescript/resolvers/tests/ts-resolvers.spec.ts

I think this is fine, as it is configured through a user-provided configuration. (https://github.com/dotansimha/graphql-code-generator/blob/d84afec0950b234cd716ea76436e3800a978328c/packages/plugins/typescript/resolvers/tests/ts-resolvers.spec.ts#L234)

**Comment by n1ru4l** [comment]
> @n1ru4l do we have any concerns about breaking changes? older TS version or anything like that? (in the generated files, i mean)

.js worked in all previous TS version afaik

**Review by saihaj — APPROVED** [review]

**Comment by n1ru4l** [comment]
Apparently, this broke Next.js... https://github.com/dotansimha/graphql-code-generator/issues/8065

**Pull request merged**